validator/client/grpc-api/grpc_validator_client.go

```diff
@@ -214,33 +219,70 @@ func NewGrpcValidatorClient(cc grpc.ClientConnInterface) iface.ValidatorClient {
 	return &grpcValidatorClient{ethpb.NewBeaconNodeValidatorClient(cc), false}
 }
 
-func (c *grpcValidatorClient) StartEventStream(ctx context.Context, topics []string, eventsChannel chan<- *eventClient.Event) {
-	ctx, span := trace.StartSpan(ctx, "validator.gRPCClient.StartEventStream")
-	defer span.End()
-	if len(topics) == 0 {
+func (c *grpcValidatorClient) streamReorg(ctx context.Context, eventsChannel chan<- *eventClient.Event) {
+	stream, err := c.beaconNodeValidatorClient.StreamReorgs(ctx, &empty.Empty{})
+	if err != nil {
 		eventsChannel <- &eventClient.Event{
-			EventType: eventClient.EventError,
-			Data:      []byte(errors.New("no topics were added").Error()),
+			EventType: eventClient.EventConnectionError,
+			Data:      []byte(errors.Wrap(client.ErrConnectionIssue, err.Error()).Error()),
 		}
 		return
 	}
-	// TODO(13563): ONLY WORKS WITH HEAD TOPIC RIGHT NOW/ONLY PROVIDES THE SLOT
-	containsHead := false
-	for i := range topics {
-		if topics[i] == eventClient.EventHead {
-			containsHead = true
-		}
-	}
-	if !containsHead {
-		eventsChannel <- &eventClient.Event{
-			EventType: eventClient.EventConnectionError,
-			Data:      []byte(errors.Wrap(client.ErrConnectionIssue, "gRPC only supports the head topic, and head topic was not passed").Error()),
+	c.isEventStreamRunning = true
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("Context canceled, stopping event stream")
+			close(eventsChannel)
```

Comment:
i'm not confident this is safe to do. I think if you close this channel here and streamHead is running it will panic if it doesn't do it in sync.